components/CharacterFilter.tsx

```diff
@@ -23,75 +24,107 @@ type CharacterFaction = {
   imageUrl: string;
 };
 
-export default function CharacterFilter(props: {
+type Props = {
   characterFilter: CharacterFilterType;
-  updateCharacterFilter: (filter: CharacterFilterType) => void;
+  updateCharacterFilter(filter: CharacterFilterType): void;
   className?: string;
   classes: { [key: string]: CharacterClass };
   factions: { [key: string]: CharacterFaction };
-}) {
+};
+
+export default function CharacterFilter({
+  characterFilter,
+  updateCharacterFilter,
+  className,
+  classes,
+  factions,
+}: Props) {
+  const activeFactionFilter =
+    characterFilter.faction !== "All" &&
+    Object.entries(factions).find(
+      ([faction]) =>
+        faction.toLowerCase() === characterFilter.faction.toLowerCase(),
+    );
+  const activeClassFilter =
+    characterFilter.class !== "All" &&
+    Object.entries(classes).find(
```

Comment:
same here, just lookup from the object using the `characterFilter.class` value as the key